tests/ftqc/test_pauli_tracker.py

```diff
@@ -186,3 +209,172 @@ def test_apply_clifford_ops_xz_value_err(self, clifford_op, xz):
             match="Please ensure xz are either 0 or 1.",
         ):
             _ = commute_clifford_op(clifford_op, xz)
+
+
+@flaky(max_runs=5)
+class TestOfflineCorrection:
+    """Tests for byproduct operation offline corrections."""
+
+    def _measurements_corrections(self, script, num_shots, raw_res):
+        """Helper function for correcting the measurement results with recorded MidMeasure results."""
+        meas_res = raw_res[0 : len(script.measurements)]
+
+        mid_meas_res = raw_res[len(script.measurements) :]
+
+        for i in range(num_shots):
+            mid_meas = [row[i] for row in mid_meas_res]
+            phase_cor = get_byproduct_corrections(script, mid_meas)
+            for j in range(len(script.measurements)):
+                meas_res[j][i] = meas_res[j][i] * phase_cor[j]
+
+        cor_res = []
+        for i in range(len(script.measurements)):
+            cor_res.append(np.sum(meas_res[i]) / num_shots)
+        return cor_res
+
+    def _get_mbqc_tape(self, ops, obs, num_shots):
+        """Helper function for creating a MBQC tape with a list of ops and obs."""
+        measurements = []
+        for ob in obs:
+            measurements.append(qml.sample(ob))
+
+        tape = qml.tape.QuantumScript(ops=ops, measurements=measurements, shots=num_shots)
+
+        (mbqc_tape,), _ = convert_to_mbqc_formalism(tape)
+        (diagonalized_mbqc_tape,), _ = diagonalize_mcms(mbqc_tape)
```

Comment:
We should be able to perform this step before/without `diagonalize_mcms` (current implementation won't work because then some `Conditional`s will be MCMs, which we can't just pass), and it should ideally be part of some transform that does these offline corrections as its post-processing function.

The easiest way to do this is going to be to make it an integral part of the `convert_to_mbqc_formalism` transform, so that you still have direct access to the MeasurementValues returned by all the operations.

Reply:
I think we might be at another point where we want to sit down and make some mock-code for what we "should" be able to do, to help with implementation details here.